packages/runtime-core/src/hydrationStrategies.ts

```diff
@@ -15,17 +15,17 @@ export type HydrationStrategy = (
   forEachElement: (cb: (el: Element) => any) => void,
 ) => (() => void) | void
 
-export type HydrationStrategyFactory<Options = any> = (
+export type HydrationStrategyFactory<Options> = (
   options?: Options,
 ) => HydrationStrategy
 
-export const hydrateOnIdle: HydrationStrategyFactory = () => hydrate => {
-  const id = requestIdleCallback(hydrate)
+export const hydrateOnIdle: HydrationStrategyFactory<number> = (timeout = 10000) => hydrate => {
+  const id = requestIdleCallback(hydrate, { timeout })
   return () => cancelIdleCallback(id)
 }
 
-export const hydrateOnVisible: HydrationStrategyFactory<string | number> =
-  (margin = 0) =>
+export const hydrateOnVisible: HydrationStrategyFactory<IntersectionObserverInit> =
+  (opts) =>
```

Comment:
Perhaps opts could have a more accurate type declaration

Reply:
A more accurate type declaration than the official typescript type for IntersectionObserver options? Do you mean making a self-contained type (in the same file) or am I misunderstanding?